### PR TITLE
Make profile of S3 store configurable

### PR DESCRIPTION
**Pull request by rbanick**

### CODE REVIEW

File: gridded_etl_tools/utils/store.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ def __init__(self, dm: dataset_manager.DatasetManager, bucket: str):
             raise ValueError("Must provide bucket name if parsing to S3")
         self.bucket = bucket
 
-    def fs(self, refresh: bool = False) -> s3fs.S3FileSystem:
+    def fs(self, refresh: bool = False, profile: str = None) -> s3fs.S3FileSystem:
```
Comment (ohsqueezy) [inline gridded_etl_tools/utils/store.py:227]: type should be `str | None`

**Review by ohsqueezy — APPROVED** [review]

File: gridded_etl_tools/utils/store.py [hunk 2]
```diff
@@ -233,14 +233,20 @@ def fs(self, refresh: bool = False) -> s3fs.S3FileSystem:
...
 
         Returns
         -------
         s3fs.S3FileSystem
             A filesystem object for interfacing with S3
         """
         if refresh or not hasattr(self, "_fs"):
-            self._fs = s3fs.S3FileSystem()
+            if profile:
+                self._fs = s3fs.S3FileSystem(profile=profile)
```
Comment (chrisrossi) [inline gridded_etl_tools/utils/store.py:247]: What would happen if we passed `profile=None` here? Do we need the if/else?

Comment (rbanick) [reply gridded_etl_tools/utils/store.py:247]: You’re right, this would simplify things

Comment (rbanick) [reply gridded_etl_tools/utils/store.py]: Zapped!

**Pull request merged**